opensearch/src/test/java/org/opensearch/sql/opensearch/data/value/OpenSearchExprValueFactoryTest.java

```diff
@@ -390,22 +401,252 @@ public void constructArray() {
             }))),
         tupleValue("{\"arrayV\":[{\"info\":\"zz\",\"author\":\"au\"}]}").get("arrayV"));
     assertEquals(
-        new ExprCollectionValue(ImmutableList.of(new ExprTupleValue(
+        new ExprCollectionValue(List.of(new ExprTupleValue(
             new LinkedHashMap<String, ExprValue>() {
               {
                 put("info", stringValue("zz"));
                 put("author", stringValue("au"));
               }
             }))),
-        constructFromObject("arrayV", ImmutableList.of(
+        constructFromObject("arrayV", List.of(
             ImmutableMap.of("info", "zz", "author", "au"))));
   }
 
   @Test
   public void constructArrayOfStrings() {
     assertEquals(new ExprCollectionValue(
-            ImmutableList.of(new ExprStringValue("zz"), new ExprStringValue("au"))),
-        constructFromObject("arrayV", ImmutableList.of("zz", "au")));
+            List.of(stringValue("zz"), stringValue("au"))),
+        constructFromObject("arrayV", List.of("zz", "au")));
+  }
+
+  @Test
+  public void constructNestedArraysOfStrings() {
+    assertEquals(
+        new ExprCollectionValue(
+            List.of(
+                collectionValue(
+                  List.of("zz", "au")
+                ),
+                collectionValue(
+                    List.of("ss")
+                )
+            )
+        ),
+        tupleValueWithArraySupport(
```

Comment:
ignore my comments below. 
what if the data is? the expected stringV is [["zz", "au"], ["ss"], ["z"]], right?
```
{ "stringV": ["z", ["s"], ["zz", "au"]] }
```

Reply:
If we execute on this:
```
tupleValueWithArraySupport("{"stringV": ["z", ["s"], ["zz", "au"]]}").get("stringV");
```

The resulting value is:
```
["z", ["s"], ["zz", "au"]]
```

In your example the order is reversed which is not the case. Does this answer your question?